src/test/java/com/shapesecurity/shift/es2016/template/FindNodesTest.java

```diff
@@ -137,19 +137,16 @@ public void testCustomMatcher() throws JsError {
 		assertEquals(info.node, firstExpr.d(Branch.BinaryExpressionRight_()).apply(tree).fromJust());
 	}
 
-	public void testCustomTypeCheck() throws JsError {
-		String source = "a + /*$ b $*/ b + /*$ c $*/ c + /*$ d $*/ d;";
+	public void testCustomPredicate() throws JsError {
+		String source = "a + /*$ b $*/ b + /*$ c $*/ c + /*$ d $*/ d + e;";
 
 		F<String, Maybe<Pair<String, Predicate<Node>>>> commentMatcher = string -> {
 			Matcher matcher = Pattern.compile("^\\$ ([^$]+) \\$$").matcher(string);
 			if (!matcher.matches()) {
 				return Maybe.empty();
 			}
 			String name = matcher.group(1);
-			return Maybe.of(Pair.of(name, node -> {
-				assertEquals(IdentifierExpression.class, node.getClass());
-				return ((IdentifierExpression) node).name.equals(name);
-			}));
+			return Maybe.of(Pair.of(name, node -> node instanceof IdentifierExpression && ((IdentifierExpression) node).name.equals(name)));
```

Comment:
Uuhh the predicate never returns `false` for this input, right? I want to see it return `false` at least once.

Reply:
For some reason I keep acting as if `+` is right-associative.

Fixed.